### PR TITLE
feat(ecs): add neuron to ecs AMI hardware type

### LINKED ISSUE #28198: aws-ecs: EcsOptimizedImage does not support Inferentia
### Describe the feature

This issue has been asked [here](https://github.com/aws/aws-cdk/issues/15862) before

[Here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-inference.html#ecs-inference-ami) is explained how to get the ECS optimized Amazon Linux 2 (Neuron) AMI. But I can't find how to use it in `ecs.AsgCapacityProvider` .

### Use Case

Using AWS Inferentia in AWS ECS.

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.105.0 (build 04cb52d)

### Environment details (OS name and version, etc.)

Ubuntu 20.04.2 LTS

### PR DESCRIPTION
**Pull request by gracelu0**
ECS now supports Amazon ECS optimized Amazon Linux 2 (Neuron) AMI variant for Amazon EC2 Inf1, Trn1 or Inf2 instances (see [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-optimized_AMI.html)). It helps make running machine learning inference workloads easier on Amazon ECS.

Add a new enum value, `NEURON` to `AMIHardwareType` to allow users to specify this AMI when setting up their ECS cluster. Add code to retrieve SSM parameter name for this AMI variant (see SSM parameter format for ECS-optimized AMI [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/retrieve-ecs-optimized_AMI.html#ecs-optimized-ami-parameter-format)).

Closes #28198 .

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs/test/integ.cluster-amazonlinux2-neuron-ami.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  enableManagedTerminationProtection: false,
+});
+
+cluster.addAsgCapacityProvider(cp);
+
+new integ.IntegTest(app, 'ClusterNeuronAmi', {
+  testCases: [stack],
+});
+
+app.synth();
```
Comment (aaythapa) [inline packages/@aws-cdk-testing/framework-integ/test/aws-ecs/test/integ.cluster-amazonlinux2-neuron-ami.ts:33]:
```suggestion
app.synth();

```

File: packages/aws-cdk-lib/aws-ecs/lib/amis.ts [hunk 2]
```diff
@@ -143,6 +148,7 @@ export class EcsOptimizedAmi implements ec2.IMachineImage {
       + (this.windowsVersion ? `Windows_Server-${this.windowsVersion}-English-Full-ECS_Optimized/` : '')
       + (this.hwType === AmiHardwareType.GPU ? 'gpu/' : '')
       + (this.hwType === AmiHardwareType.ARM ? 'arm64/' : '')
+      + (this.hwType === AmiHardwareType.NEURON ? 'inf/' : '')
```
Comment (aaythapa) [inline packages/aws-cdk-lib/aws-ecs/lib/amis.ts:151]: Should we test this change in the unit tests or is there no way to do that?

**Review by aaythapa — COMMENTED** [review]
Looks good other than some minor comments

File: packages/aws-cdk-lib/aws-ecs/lib/amis.ts (hunk 2 above)
Comment (gracelu0) [reply packages/aws-cdk-lib/aws-ecs/lib/amis.ts:151]: Added a unit test to check the template has the correct SSM parameter

**Review by aaythapa — APPROVED** [review]

**Pull request merged**